lib/alchemy/upgrader/three_point_one.rb

```diff
@@ -1,8 +1,41 @@
+require 'thor'
+
+class Alchemy::Upgrader::ThreePointOneTask < Thor
+  include Thor::Actions
+
+  no_tasks do
+
+    def patch_acts_as_taggable_on_migrations
+      sentinel = /def self.up/
+
+      aato_file = Dir.glob('db/migrate/*_acts_as_taggable_on_migration.*.rb').first
+      if aato_file
+        inject_into_file aato_file,
+          "\n    # inserted by Alchemy CMS upgrader\n    return if table_exists?(:tags)\n",
+          { after: sentinel, verbose: true }
+      end
+
+      aato_file = Dir.glob('db/migrate/*_add_missing_unique_indices.*.rb').first
+      if aato_file
+        inject_into_file aato_file,
+          "\n    # inserted by Alchemy CMS upgrader\n    return if index_exists?(:tags, :name)\n",
+          { after: sentinel, verbose: true }
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Redundant curly braces around a hash parameter.<br>Space inside { detected.<br>Space inside } detected.